src/room_member.cpp

```diff
@@ -256,12 +256,24 @@ bool RoomMember::IsConnected() const {
     return state == State::Joining || state == State::Joined;
 }
 
+void RoomMember::CloseConnection() {
+    peer->Shutdown(300);
+    member_information.clear();
+    nickname.clear();
+    mac_address = {};
+    room_information = {};
+
+    chat_queue.clear();
```

Comment:
My thinking was if you leave a room while your queues aren't empty and join a new one, before the user calls them, they 'live' in the wrong room. Maybe clearing them on join?